packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart

```diff
@@ -0,0 +1,204 @@
+  });
+
+  tearDown(() {
+    tryToDelete(tempDir);
+  });
+
+  testUsingContext('generated plugin registrant passes analysis', () async {
+    await _createProject(projectDir, <String>[]);
+    // We need to add a dependency with web support to trigger
+    // the generated_plugin_registrant generation.
+    await _addDependency(projectDir, 'shared_preferences', '^0.5.12+4');
+    await _analyzeProject(projectDir);
+
+    expect(
+        projectDir.childFile('lib/generated_plugin_registrant.dart'), exists);
+  }, overrides: <Type, Generator>{
+    Pub: () => Pub(
+          fileSystem: globals.fs,
+          logger: globals.logger,
+          processManager: globals.processManager,
+          usage: globals.flutterUsage,
+          botDetector: globals.botDetector,
+          platform: globals.platform,
+        ),
+  });
+
+  testUsingContext(
+      'generated plugin registrant ignores lines longer than 80 chars',
+      () async {
+    await _createProject(projectDir, <String>[]);
+    await _addAnalysisOptions(
+        projectDir, <String>['lines_longer_than_80_chars']);
+    await _createProject(tempDir.childDirectory('test_plugin'), <String>[
+      '--template=plugin',
+      '--platforms=web',
+      '--project-name',
+      'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+    ]);
+    // The line for the test web plugin (`  TestWebPluginWithAPurposefullyExtremelyLongPackageNameWeb.registerWith(registrar);`)
+    // exceeds 80 chars.
+    // With the above lint rule added, we want to ensure that the `generated_plugin_registrant.dart`
+    // file does not fail analysis (this is a regression test - an ignore was
+    // added to cover this case).
+    await _addDependency(
+        projectDir,
+        'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+        '\n        path: ../test_plugin');
+    await _analyzeProject(projectDir);
+
+    expect(
+        projectDir.childFile('lib/generated_plugin_registrant.dart'), exists);
+  }, overrides: <Type, Generator>{
+    Pub: () => Pub(
+          fileSystem: globals.fs,
+          logger: globals.logger,
+          processManager: globals.processManager,
+          usage: globals.flutterUsage,
+          botDetector: globals.botDetector,
+          platform: globals.platform,
+        ),
+  });
+}
+
+Future<void> _ensureFlutterToolsSnapshot() async {
+  final String flutterToolsPath = globals.fs.path.absolute(globals.fs.path.join(
+    'bin',
+    'flutter_tools.dart',
+  ));
+  final String flutterToolsSnapshotPath =
+      globals.fs.path.absolute(globals.fs.path.join(
+    '..',
+    '..',
+    'bin',
+    'cache',
+    'flutter_tools.snapshot',
+  ));
+  final String dotPackages = globals.fs.path.absolute(globals.fs.path.join(
+    '.packages',
+  ));
+
+  final File snapshotFile = globals.fs.file(flutterToolsSnapshotPath);
+  if (snapshotFile.existsSync()) {
+    snapshotFile.renameSync(flutterToolsSnapshotPath + '.bak');
```

Comment:
this seems to just create a file and then leave it there?

Reply:
I am actually not sure about this 🤔 

I found this in `create_test` and I included it here because I also used the analysis function, which seems to depend on `flutter_tools.snapshot`. Here is the origin: https://github.com/flutter/flutter/pull/39338

Reply:
cc @zanderso

Reply:
See https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/permeable/create_test.dart#L73

Reply:
@zanderso Oh, I see! I will add that part 👍